build.sh

```diff
@@ -62,6 +84,13 @@ for PKG_PATH in $(catkin_topological_order --only-folders); do
   test -f "$PKG_PATH/COLCON_IGNORE" && echo "Skipped" && continue
   (
   cd "$PKG_PATH"
+  test -f debian/changelog && GENERATED_DEBIAN_PACKAGE=$(head -n1 debian/changelog | cut -d' ' -f1)
+  if [ -n "$CONTINUE_PACKAGE_GENERATION" ] && [ -n "$GENERATED_DEBIAN_PACKAGE" ] && ls "${APT_REPO}/${GENERATED_DEBIAN_PACKAGE}_"*.deb >/dev/null 2>&1; then
+    echo " Skipping already generated package: ${GENERATED_DEBIAN_PACKAGE}"
+    COUNT=$((COUNT+1))
+    echo "::endgroup::"
+    continue
```

Comment:
continue does not work in a subshell.

Reply:
It seems to be working fine locally, though:

```shell
...
+ test -f src/universe/autoware.universe/system/default_ad_api_helpers/ad_api_visualizers/CATKIN_IGNORE
+ test -f src/universe/autoware.universe/system/default_ad_api_helpers/ad_api_visualizers/COLCON_IGNORE
+ cd src/universe/autoware.universe/system/default_ad_api_helpers/ad_api_visualizers
+ test -f debian/changelog
+ head -n1 debian/changelog
+ cut -d  -f1
+ GENERATED_DEBIAN_PACKAGE=ros-humble-ad-api-visualizers
+ [ -n 1 ]
+ [ -n ros-humble-ad-api-visualizers ]
+ ls /home/runner/ros-deb-builder-action/apt_repo/ros-humble-ad-api-visualizers_0-2023.02.09.15.40_amd64.deb
+ echo  Skipping already generated package: ros-humble-ad-api-visualizers
 Skipping already generated package: ros-humble-ad-api-visualizers
+ COUNT=2
+ echo ::endgroup::
::endgroup::
+ continue
+ COUNT=2
+ echo ::endgroup::
::endgroup::
+ echo ::group::Building 2/235: src/core/autoware_adapi_msgs/autoware_adapi_v1_msgs
::group::Building 2/235: src/core/autoware_adapi_msgs/autoware_adapi_v1_msgs
+ test -f src/core/autoware_adapi_msgs/autoware_adapi_v1_msgs/CATKIN_IGNORE
+ test -f src/core/autoware_adapi_msgs/autoware_adapi_v1_msgs/COLCON_IGNORE
+ cd src/core/autoware_adapi_msgs/autoware_adapi_v1_msgs
+ test -f debian/changelog
+ head -n1 debian/changelog
+ cut -d  -f1
+ GENERATED_DEBIAN_PACKAGE=ros-humble-autoware-adapi-v1-msgs
+ [ -n 1 ]
+ [ -n ros-humble-autoware-adapi-v1-msgs ]
+ ls /home/runner/ros-deb-builder-action/apt_repo/ros-humble-autoware-adapi-v1-msgs_0-2023.02.09.15.41_amd64.deb
+ echo  Skipping already generated package: ros-humble-autoware-adapi-v1-msgs
 Skipping already generated package: ros-humble-autoware-adapi-v1-msgs
+ COUNT=3
...
```

or you mean something else?

Reply:
Afaik that depends on the shell implementation and `exit` should always be correct.

Reply:
As you surely, `/bin/sh` on Debian points to `dash`, which runs as POSIX dash, so if this passes with `/bin/sh`, it should be safe to assume that `continue` is fine, wouldn't it? I might be mistaken here, though.

Reply:
What's wrong with using exit here?